src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -143,14 +143,21 @@ DatabaseShardImp::init()
                     j_)};
                 if (!shard->open(scheduler_, *ctx_))
                 {
-                    if (!shard->isLegacy())
-                        return false;
-
-                    // Remove legacy shard
                     shard->removeOnDestroy();
-                    JLOG(j_.warn()) <<
-                        "shard " << shardIndex <<
-                        " removed, legacy shard";
+                    if (!shard->isLegacy())
+                    {
+                        // Remove corrupted shard
+                        JLOG(j_.warn()) <<
+                            "shard " << shardIndex <<
+                            " removed, corrupted shard";
+                    }
+                    else
+                    {
+                        // Remove legacy shard
+                        JLOG(j_.warn()) <<
```

Comment:
Could you refactor these two log messages into a single statement? You can use a ternary operator to pick whether to output "corrupted" or "legacy".

Reply:
Done